arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/fragment/common/CommonFragmentValues.java

```diff
@@ -186,4 +194,56 @@ static int computePcNew(
 
     return pc + 1;
   }
+
+  private long computeGasRemaining() {
+    return hub.remainingGas();
+  }
+
+  public long computeGasExpected() {
+
+    final CallFrame currentFrame = hub.currentFrame();
+
+    if (hub.state().getProcessingPhase() != TX_EXEC) return 0;
+
+    if (currentFrame.executionPaused()) {
+      currentFrame.unpauseCurrentFrame();
+      return currentFrame.lastValidGasNext();
+    }
+
+    return currentFrame.frame().getRemainingGas();
+  }
+
+  private long computeGasCost() {
+    return Hub.GAS_PROJECTOR.of(hub.messageFrame(), hub.opCode()).upfrontGasCost();
+  }
+
```

Comment:
rename to `computeUpfrontGasCost`?

Reply:
It's a little strange, I agree. But we have an issue here with naming:
- `GAS_COST` in the spec typically contains the `upfrontGasCost`
- the "real gas cost" (e.g. for CALL's and CREATE's) is deduced from that and other elements
- things get even more annoying for aborted CALL's, EOA CALL's, precompile CALL's

I don't think we have a good means to solve this. I would suggest keeping `GAS_COST` as a name or renaming it to `GAS_UPFRONT`. We can change this later.